Older versions of zeroize (<1.5.0) don't work with Rust 1.58

When trying to compile the `x25519-dalek-ng` crate with Rust 1.58, I get the following errors:

```
error[E0405]: cannot find trait `ZeroizeOnDrop` in crate `zeroize`
  --> src/x25519.rs:64:10
   |
64 | #[derive(Zeroize)]
   |          ^^^^^^^ not found in `zeroize`
   |
   = note: this error originates in the derive macro `Zeroize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0405]: cannot find trait `ZeroizeOnDrop` in crate `zeroize`
   --> src/x25519.rs:112:17
    |
112 | #[derive(Clone, Zeroize)]
    |                 ^^^^^^^ not found in `zeroize`
    |
    = note: this error originates in the derive macro `Zeroize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0405]: cannot find trait `ZeroizeOnDrop` in crate `zeroize`
   --> src/x25519.rs:158:10
    |
158 | #[derive(Zeroize)]
    |          ^^^^^^^ not found in `zeroize`
    |
    = note: this error originates in the derive macro `Zeroize` (in Nightly builds, run with -Z macro-backtrace for more info)
```

The crate depend on zeroize 1.2.0, but updating it to 1.4.x doesn't help. Only 1.5.0 seems to work.

--- a/zeroize/derive/src/lib.rs
+++ b/zeroize/derive/src/lib.rs
@@ -317,8 +317,6 @@ fn derive_zeroize_with_drop(s: synstructure::Structure<'_>) -> TokenStream {
         }
     });
 
-    let zeroize_on_drop_impl = impl_zeroize_on_drop(&s);
-
     let zeroize_impl = derive_zeroize_without_drop(s);
 
     quote! {
@@ -326,8 +324,6 @@ fn derive_zeroize_with_drop(s: synstructure::Structure<'_>) -> TokenStream {
 
         #[doc(hidden)]
         #drop_impl
-
-        #zeroize_on_drop_impl
     }
 }
 
@@ -417,12 +413,6 @@ mod tests {
                             self.zeroize();
                         }
                     }
-                };
-                #[allow(non_upper_case_globals)]
-                #[doc(hidden)]
-                const _DERIVE_zeroize_ZeroizeOnDrop_FOR_Z: () = {
-                    extern crate zeroize;
-                    impl zeroize::ZeroizeOnDrop for Z {}
                 };
             }
             no_build // tests the code compiles are in the `zeroize` crate

